### PR TITLE
[prometheus-node-exporter] add version value to override image.tag

### LINKED ISSUE #3821: [prometheus-node-exporter] Using tag 'latest' breaks prometheus-node-exporter
When using an image with the tag `latest`, the [semverCompare function in the template does not work properly](https://github.com/prometheus-community/helm-charts/blob/1a60cd7555e3f1a0239c28aee04a767a571a6ad9/charts/prometheus-node-exporter/templates/daemonset.yaml#L53).  

Recommend change the defaultValue function to only look at the image tag if it matches proper SemVer formatting.

### LINKED ISSUE #4104: [prometheus-node-exporter] Using custom image tags breaks chart
### Describe the bug a clear and concise description of what the bug is.

My organization requires all 3rd-party images to be imported into our registry for security reasons, and all 3rd-party images are tagged with the original complete image name sanitized (e.g `quay.io_prometheus_node-exporter_v1.7.0`)
Using this image tag breaks the chart with 

```
Error: template: kube-prometheus-stack/charts/prometheus-node-exporter/templates/daemonset.yaml:53:19: executing "kube-prometheus-stack/charts/prometheus-node-exporter/templates/daemonset.yaml" at <semverCompare ">=1.4.0" (default .Chart.AppVersion .Values.image.tag)>: error calling semverCompare: Invalid Semantic Version
```

## Suggested solution

Add a value to values.yaml named `version` and use that value (if exists) instead of `image.tag`, like in [prometheus-operator chart](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml#L57C1-L57C111)

```
  version: {{ default .Values.prometheus.prometheusSpec.image.tag .Values.prometheus.prometheusSpec.version }}
```

### What's your helm version?

version.BuildInfo{Version:"v3.13.3", GitCommit:"c8b948945e52abba22ff885446a1486cb5fd3474", GitTreeState:"clean", GoVersion:"go1.20.11"}

### What's your kubectl version?

WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version. Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.17", GitCommit:"22a9682c8fe855c321be75c5faacde343f909b04", GitTreeState:"clean", BuildDate:"2023-08-23T23:44:35Z", GoVersion:"go1.20.7", Compiler:"gc", Platform:"linux/amd64"} Kustomize Version: v4.5.4 Server Version: version.Info{Major:"1", Minor:"24+", GitVersion:"v1.24.17-eks-8cb36c9", GitCommit:"b114b128d961e18fed9242aed57f37f289bc740f", GitTreeState:"clean", BuildDate:"2023-11-22T21:53:08Z", GoVersion:"go1.20.10", Compiler:"gc", Platform:"linux/amd64"}

### Which chart?

prometheus-node-exporter

### What's the chart version?

4.24.0

### What happened?

_No response_

### What you expected to happen?

_No response_

### How to reproduce it?

_No response_

### Enter the changed values of values.yaml?

```
image:
  registry: <REDACTED>
  repository: <REDACTED>
  tag: quay.io_prometheus_node-exporter_v1.7.0
```

### Enter the command that you execute and failing/misfunctioning.

`helm template .`

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by mohamed-essam**
#### What this PR does / why we need it

Adds a new value `.Values.version` to override reading version from `.Values.image.tag` or `.Chart.AppVersion` in case of custom repositories.

#### Which issue this PR fixes

- fixes #4104
- fixes #3821 

#### Special notes for your reviewer

N/A

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by zeritti — CHANGES_REQUESTED** [review]
Thank you, @mohamed-essam, for your PR. Please, see my comments below.

File: charts/prometheus-node-exporter/Chart.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ keywords:
   - prometheus
   - exporter
 type: application
-version: 4.24.0
+version: 4.24.1
```
[the remark this answers is not in the document]
Comment (mohamed-essam) [reply charts/prometheus-node-exporter/Chart.yaml]: Understandable, I considered it a minor fix as it solved a somewhat niche problem and wouldn't cause any issues with backward compatibility

File: charts/prometheus-node-exporter/templates/daemonset.yaml [hunk 2]
```diff
@@ -50,7 +50,7 @@ spec:
             - --path.sysfs=/host/sys
             {{- if .Values.hostRootFsMount.enabled }}
             - --path.rootfs=/host/root
-            {{- if semverCompare ">=1.4.0" (default .Chart.AppVersion .Values.image.tag) }}
+            {{- if semverCompare ">=1.4.0" (.Values.version | default .Values.image.tag | default .Chart.AppVersion) }}
```
[the remark this answers is not in the document]
Comment (mohamed-essam) [reply charts/prometheus-node-exporter/templates/daemonset.yaml]: Didn't know this function existed, thank you :pray:

File: charts/prometheus-node-exporter/values.yaml [hunk 3]
```diff
@@ -478,3 +478,6 @@ extraManifests: []
   #     name: prometheus-extra
   #   data:
   #     extra-data: "value"
+
+# Override version of app, required if image.tag is defined and does not follow semver
+version: ""
```
Comment (zeritti) [inline charts/prometheus-node-exporter/values.yaml:483]: Please, fix the linting error (missing new line character at the end of the file).

**Review by zeritti — CHANGES_REQUESTED** [review]

File: charts/prometheus-node-exporter/values.yaml (hunk 3 above)
Comment (mohamed-essam) [reply charts/prometheus-node-exporter/values.yaml]: Weird it only showed up now though...

**Review by zeritti — APPROVED** [review]

**Comment by mohamed-essam** [comment]
@zeritti Any chance to get this merged soon? As I will still need to open another PR to upgrade the dependency in kube-prometheus-stack for this

**Comment by zeritti** [comment]
@zanhsieh, please, have a look.

**Pull request merged**